kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -269,14 +270,79 @@ def _local_url_parse(id_value, version_params):
             query_params = parse_qs(parsed_id.query)
 
         if version_params is not None:
-            if not isinstance(query_params, dict):
-                query_params = {}
             query_params.update(version_params)
 
-        id_without_query = f"{parsed_id.scheme}://{parsed_id.netloc}{parsed_id.path}{'#' + fragment if fragment is not None else ''}"
+        formatted_fragment = '#' + fragment if fragment is not None else ''
+        id_without_query = f"{parsed_id.scheme}://{parsed_id.netloc}{parsed_id.path}{formatted_fragment}"
 
         return id_without_query, query_params
 
+    def _retrieve_id(
+            self,  id_, retrieve_source: bool, cross_bucket: bool, query_params: Dict
+    ):
+
+        url_base = self.service.url_resolver if cross_bucket else self.service.url_resources
+
+        url_resource = Service.add_schema_and_id_to_endpoint(
+            url_base, schema_id=None, resource_id=id_
+        )
+        # and not cross_bucket: if cross_bucket, no support for /source and metadata.
+        # So this will fetch the right metadata. The source data will be fetched later
+        url = f"{url_resource}/source" if retrieve_source and not cross_bucket else url_resource
+
+        response = requests.get(
+            url, params=query_params, headers=self.service.headers, timeout=REQUEST_TIMEOUT
+        )
+        catch_http_error_nexus(response, RetrievalError)
+
+        try:
+            data = response.json()
+            resource = self.service.to_resource(data)
+        except Exception as e:
+            raise RetrievalError(e) from e
+
+        # retrieve_source = False = metadata in payload
```

Comment:
why do you keep this commented?

Reply:
is this like a little documentation?

Reply:
Yeah I'm temporarily keeping track of the old logic and the new one as a way to explain the new implementation. I could remove it though

Reply:
you could keep it as a docstring of `_retrieve_id`, no?

Reply:
Yes I can definitely change it